File: conversions/omit/elide-omit.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

puts "=== Ruby Calling TypeScript Omit ==="
puts

# Example: Rails API Sanitization
# omit_module = Elide.require('./elide-omit.ts')
#
# class UsersController < ApplicationController
#   def show
#     user = User.find(params[:id])
#     user_hash = user.as_json
#     safe_user = omit_module.call(:default, user_hash, 'password', 'salt', 'api_secret')
#     render json: safe_user
#   end
# end

# Example: Sidekiq Job Data Sanitization
# class DataProcessorJob
#   include Sidekiq::Job
#
#   def perform(user_data)
#     # Remove sensitive fields before logging
#     safe_data = omit_module.call(:default, user_data, 'password', 'ssn', 'credit_card')
#     logger.info "Processing user: #{safe_data}"
#     process_data(user_data)
#   end
# end

puts "Real-world use case:"
puts "- Ruby Rails/Sidekiq apps sanitize sensitive data"
puts "- Uses same TypeScript implementation as Node.js service"
puts "- Guarantees consistent security across entire stack"
puts "- No need to install Ruby gem"
puts

puts "When Elide Ruby API is ready, usage will be:"
puts "  omit_module = Elide.require('./elide-omit.ts')"
puts "  safe = omit_module.call(:default, user, 'password', 'salt')  # That's it!"

File: converted/utilities/omit/elide-omit.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-omit

This demonstrates calling the TypeScript omit implementation from Python
using Elide's polyglot capabilities.

Benefits:
- One omit implementation shared across TypeScript and Python
- Consistent data sanitization across services
- No Python library needed
- Guaranteed security consistency
"""

print("=== Python Calling TypeScript Omit ===\n")

# Example: API Response Sanitization
# user_data = {
#     'id': 123,
#     'username': 'alice',
#     'email': 'alice@example.com',
#     'password': 'hashed_password',
#     'salt': 'random_salt',
#     'role': 'user'
# }
#
# from elide import require
# omit_module = require('./elide-omit.ts')
# safe_data = omit_module.default(user_data, 'password', 'salt')
# print(f"Safe data: {safe_data}")

# Example: Flask API Sanitization
# from flask import Flask, jsonify
# app = Flask(__name__)
#
# @app.route('/api/user/<int:user_id>')
# def get_user(user_id):
#     user = db.get_user(user_id)  # Full user record
#     safe_user = omit_module.default(user, 'password', 'salt', 'apiKey')
#     return jsonify(safe_user)

# Example: Django View Sanitization
# from django.http import JsonResponse
#
# def user_detail(request, user_id):
#     user = User.objects.get(id=user_id)
#     user_dict = model_to_dict(user)
#     safe_user = omit_module.default(user_dict, 'password', 'secret_token')
#     return JsonResponse(safe_user)

print("Real-world use case:")
print("- Python Flask/Django apps sanitize sensitive data")
print("- Uses same TypeScript implementation as Node.js service")
print("- Guarantees consistent security across entire stack")
print("- No need to install Python omit library")
print()

print("Problem Solved:")
print("Before: Manual dict comprehension = easy to forget sensitive fields")
print("After: One Elide implementation = consistent sanitization")
print()

print("When Elide Python API is ready, usage will be:")
print("  from elide import require")
print("  omit = require('./elide-omit.ts')")
print("  safe = omit.default(user, 'password', 'salt')  # That's it!")
